### ISSUE TITLE
Non global logger and statistics callback

### ISSUE THREAD

**Issue by mensfeld** [appsignal/rdkafka-ruby#82]
Majority of the settings of this lib are per instance based.

Is there any reason why `logger` and `statistics_callback` aren't? It is really problematic for systems where you may have several instances of rdkafka running for producing and consuming with different policies around instrumentation.

**Comment by thijsc** [comment]
Looking at the docs it does seem like both are scoped on the client. So it should be possible to move them to a config's instance. I have some other issues with higher priority, will take a look at this after those. Also open to a pull of course.

**Comment by thijsc** [comment]
Closing this one since we don't currently need this. If anybody needs this do reopen.

**Comment by mensfeld** [comment]
> If anybody needs this do reopen.

I do not have a way to reopen :(

**Comment by thijsc** [comment]
Are you interested in working on this?

**Comment by mensfeld** [comment]
@thijsc yes. Just leave it open for 2-3 more weeks. I'm finishing Karafka 2.0 and then I will have some time to adjust things on rdkafka side :)

**Comment by mensfeld** [comment]
Closing as I was able to bypass this limitation without changing the design. If anyone needs this, just ping me.

**Comment by trickleup** [comment]
@mensfeld - running into the same issue that you describe, of needing to have multiple instances of producers/consumers, and to be able to isolate the statistics_callbacks between those instances.

> I was able to bypass this limitation without changing the design. If anyone needs this, just ping me.

Do you mind sharing the gist of what you did to achieve this?

**Comment by mensfeld** [comment]
@trickleup I'm on a holiday and I plan to backport if from karafka and waterdrop to rdkafka once back (2 weeks). 

If you are looking for multiple instances I would recommend going with the wrapper I did waterdrop: https://github.com/karafka/waterdrop as it tackles those cases. The code for this is also somewhere there. Writing from mobile so can't locate the exact files sorry :pray:

**Comment by jychen7** [comment]
> I plan to backport if from karafka and waterdrop to rdkafka once back (2 weeks).

👍

**Comment by mensfeld** [comment]
@jychen7 ok clearly it is taking me more than that but my original solution still stands and I still have plans to merge this and other things to the upstream
